horizon/enforcer/api.py

```diff
@@ -162,6 +182,41 @@ def get_v2_processed_query(result: dict) -> Optional[dict]:
     return result.get("debug", {}).get("input", None)
 
 
+async def notify_seen_sdk(
```

Comment:
I like the idea of this being a dependency, but don't you want to just call it from the main router instead of in every route?

Reply:
which main router ?